### DISCUSSION TITLE
How do I customize an anchor at some title in a MD file so that I can link to it from the file itself and other files?

### DISCUSSION THREAD

**Discussion by GarfieldJiang** [mkdocs/mkdocs#3754] [Q&A] [answered]
I know things like `../my.md#hello-world` but my problem is that it doesn't work well with non-ascii characters in the title (like Chinese characters).

**Comment by Andre601** [comment, marked as answer]
Not sure what you exactly want, but using the `attr_list` extension should allow you to override the header like this:
```markdown
## Lorem Ipsum { #dolor-sit-amet }
```
The above header would show `Lorem Ipsum` while the ID would be `#dolor-sit-amet`

**Reply by GarfieldJiang** [reply]
Thanks I'll give it a try.

**Reply by tunggnu** [reply]
This solution is very simple and it solved my problem. Thank you.

**Reply by pp-mluthi** [reply]
Hi,
I'm new with markdown and mkdocs and having the same error generated when using custom anchors

Having written this markdown :
```
- [New card] (#create-new-card)

## Create a new card {#create-new-card}
```

The error is `Doc file contains a link '#create-new-card', but there is no such anchor on this page.` and the generated HTML id is 'new-card-create-new-card'.

What did I do wrong ? ;)

Thanks!

**Reply by kamilkrzyskow** [reply]
Hi @pp-mluthi 
You probably forgot to enable the `attr_list` extension, here's an example: 

[3754-mkdocs.zip](https://github.com/user-attachments/files/21761144/3754-mkdocs.zip)

https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#attribute-lists

I'm not sure if this is enabled with [default validation](https://www.mkdocs.org/user-guide/configuration/#validation), because I only tried with full strictness, but keep in mind that there are some cases with false positives for the current validation implementation:
- https://github.com/mkdocs/mkdocs/issues/3690
